trestle/core/profile_resolver.py

```diff
@@ -0,0 +1,650 @@
+# Copyright (c) 2020 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Create ssp from catalog and profile."""
+
+import logging
+import pathlib
+import re
+from typing import Any, Dict, Iterable, List, Optional, Set, Tuple, Union
+from uuid import uuid4
+
+from pydantic import BaseModel
+
+import trestle.oscal.catalog as cat
+import trestle.oscal.profile as prof
+from trestle.core.const import MARKDOWN_URL_REGEX, UUID_REGEX
+from trestle.core.err import TrestleError
+from trestle.core.pipeline import Pipeline
+from trestle.core.remote import cache
+from trestle.oscal import common
+
+logger = logging.getLogger(__name__)
+
+
+class PathNode(BaseModel):
+    """Node in path to control in catalog."""
+
+    id_: str
+    is_control: bool
```

Comment:
I had already decided that the whole PathNode structure was overkill since groups are only at the beginning of a path - so I have removed it entirely.

Reply:
Perfect!